### PR TITLE
Add v2.5.0 CHANGELOG

### PR DESCRIPTION
**Pull request by msau42**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>

/kind documentation

**What this PR does / why we need it**:
We forgot to update the changelog for the last release. Copying and filtering the list in the release note.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+* Update Klog to V2 by @Kartik494 in https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/234
+* Duplicate volume guard by @davidmccormick in https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/246
+* add local volume discovery period flag by @dabaooline in https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/261
+* Multi linux arch and multi windows distro builds by @mauriciopoppe in https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/273
+* Update discovery and deletion code to work in Windows nodes through CSI Proxy by @mauriciopoppe in https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/276
+* feat: add lstc2022 windows image build by @andyzhangx in https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/314
+* fix: golang.org/x/crypto for CVE-2022-27191 by @andyzhangx in https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/321
+* feat: support namePattern as a list by @andyzhangx in https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/320
+
+Deployment updates:
```
Comment (msau42) [inline CHANGELOG.md:17]: @mauriciopoppe not sure where you want to put helm chart release notes now that it's separate.

Comment (mauriciopoppe) [reply CHANGELOG.md:17]: I'll create a PR to add a CHANGELOG.md in the helm chart root.

**Comment by mauriciopoppe** [comment]
I think you need to sync to master to fix the spelling error

**Pull request merged**